packages/flutter_tools/lib/src/commands/precache.dart

```diff
@@ -74,11 +74,14 @@ class PrecacheCommand extends FlutterCommand {
       if (artifact.feature != null && !featureFlags.isEnabled(artifact.feature)) {
         continue;
       }
-      if (boolArg(artifact.name)) {
-        requiredArtifacts.add(artifact);
-      }
       // The `android` flag expands to android_gen_snapshot, android_maven, android_internal_build.
-      if (artifact.name.startsWith('android_') && boolArg('android')) {
+      if (artifact.name.startsWith('android_')) {
```

Comment:
I don't _think_ we can detect that --android_* was explicitly added vs just excluded, since it's true by default.  That was the problem in my case: The tool was parsing `flutter precache --no-android` as `flutter precache --no-android --android_gen_snapshot --android_maven --android_internal_build`

Reply:
> argResults.arguments

Guess it would help if I close-read your comment...
Good idea. 😄